docs/config-walkthrough.md

```diff
@@ -0,0 +1,222 @@
+Configuration Walkthroughs
+==========================
+
+*If you're looking for reference documentation on configuration, please
+read the the [configuration reference](/docs/config.md)*
+
+Per-pod HTTP Requests
+---------------------
+
+### Background
+
+*The [full walkthrough](/docs/walkthrough.md) sets up a the background for
+something like this*
+
+Suppose we have some frontend webserver, and we're trying to write an
+configuration for the Promtheus adapter so that we can autoscale it based
+on the HTTP requests per second that it receives.
+
+Before starting, we've gone and instrumented our frontend server with
+a metric, `http_requests_total`.  It is exposed with a single label,
+`method`, breaking down the requests by HTTP verb.
+
+We've configured our Prometheus to collect the metric, and our promethues
+adds the `kubernetes_namespace` and `kubernetes_pod_name` labels,
+representing namespace and pod, respectively.
+
+If we query Prometheus, we see series that look like
+
+```
+http_requests_total{method="GET",kubernetes_namespace="production",kubernetes_pod_name="frontend-server-abcd-0123"}
+```
+
+### Configuring the adapter
+
+The adapter considers metrics in the following ways:
+
+1. First, It discovers the metrics available (*Discovery*)
+
+2. Then, it figures out which Kubernetes resources each metric is
+   associated with (*Association*)
+
+3. Then, it figures out how it should expose them to the custom metrics
+   API (*Naming*)
+
+4. Finally, it figures out how it should query Prometheus to get the
+   actual numbers (*Querying*)
+
+We need to inform the adapter how it should perform each of these steps
+for our metric, `http_requests_total`, so we'll need to add a new
+***rule***. Each rule in the adapter encodes these steps.  Let's add a new
+one to our configuration:
+
+```yaml
+rules:
+- {}
+```
+
+If we want to find all `http_requests_total` series ourselves in the
+Prometheus dashboard, we'd write
+`http_requests_total{kubernetes_namespace!="",kubernetes_pod_name!=""}` to
+find all find all `http_requests_total` series that were associated with
+a namespace and pod.
+
+We can add this to our rule in the `seriesQuery` field, to tell the
+adapter how *discover* the right series itself:
+
+```yaml
+rules:
+- seriesQuery: 'http_requests_total{kubernetes_namespace!="",kubernetes_pod_name!=""}'
+```
+
+Next, we'll need to tell the adapter how to figure out which Kubernetes
+resources are associated with the metric.  We've already said that
+`kubernetes_namespace` represents the namespace name, and
+`kubernetes_pod_name` represents the pod name.  Since these names don't
+quite follow a consistent pattern, we use the `overrides` section of the
+`resources` field in our rule:
+
+```yaml
+rules:
+- seriesQuery: 'http_requests_total{kubernetes_namespace!="",kubernetes_pod_name!=""}'
+  resources:
+    overrides:
+      kubernetes_namespace: {resource: "namespace"}
```

Comment:
Is there somewhere that lists the available resource types? It might be good to include a link to that around this section.

Reply:
There is a command, but it can change at any time due to CRDs, aggregated API servers, etc.  I'll add a note about it.